cargo-dist/src/errors.rs

```diff
@@ -375,6 +375,18 @@ pub enum DistError {
         /// The input
         pair: String,
     },
+
+    /// Checksum mismatch when fetching vendored actions
+    #[error("Checksum mismatch when fetching vendored GitHub action {action}")]
+    #[diagnostic(help("Expected {actual} but got {expected}"))]
+    VendoredActionHashMismatch {
+        /// The checksum we expected
+        expected: String,
+        /// The actual checksum
+        actual: String,
+        /// The action we were fetching (owner/repo format)
+        action: String,
```

Comment:
we might call this owner_repo (or just repo as it's called above?) because action was definitely confusing to me.